packages/text-bitmap/src/BitmapFontFactory.ts

```diff
@@ -0,0 +1,221 @@
+import { BitmapFont } from './BitmapFont';
+import { TextMetrics, TextStyle } from '@pixi/text';
+import type { ITextStyle } from '@pixi/text';
+import { generateFillStyle } from './utils/generateFillStyle';
+import { Texture, BaseTexture } from '@pixi/core';
+import { BitmapFontData } from './BitmapFontData';
+import { Rectangle } from '@pixi/math';
+import { hex2rgb, string2hex } from '@pixi/utils';
+
+const BMT_SIZE = 1024;
+const BMT_PADDING = 4;
+
+/**
+ * Factory for generating bitmap fonts at runtime.
+ *
+ * @memberof PIXI
+ * @class
+ */
+export class BitmapFontFactory
+{
+    /**
+     * Generates a bitmap-font for the given style and character set. This does not support
+     * kernings yet.
+     *
+     * @param {PIXI.IBitmapFontFactoryOptions} options
+     */
+    public static create(options: Partial<IBitmapFontFactoryOptions> = {}): BitmapFont
```

Comment:
Then you don't need Partial here and options should be non-optional.
```suggestion
    public static create(options: IBitmapFontFactoryOptions): BitmapFont
```